app/models/survey/comment.rb

```diff
@@ -0,0 +1,17 @@
+class Survey::Comment < ApplicationRecord
+  belongs_to :poll, class_name: "Survey::Poll", optional: true, foreign_key: "survey_poll_id"
+
+  scope :visible, -> { where(visible: true) }
+  scope :invisible, -> { where(visible: false) }
+  scope :filtered_for_current_user, lambda { |current_user|
+    return all if current_user.admin_role?
+    return visible if current_user.app_role?
+
+    if current_user.editor_role?
+      data_provider_ids = [current_user.data_provider_id] + User.restricted_role.map(&:data_provider_id)
+      return includes(:poll).where(poll: { data_provider_id: data_provider_ids.compact.flatten })
+    end
+
+    includes(:poll).where(survey_polls: { data_provider_id: current_user.data_provider_id })
+  }
```

Comment:
wieso nicht `include FilterByRole`, um die scopes automatisch zu haben?

Reply:
ich sehe einen unterschied bei `filtered_for_current_user`, der überschrieben werden könnte hier und die andern kommen über den include, oder?

Reply:
so?

```
class Survey::Comment < ApplicationRecord
  include FilterByRole

  belongs_to :poll, class_name: "Survey::Poll", optional: true, foreign_key: "survey_poll_id"

  scope :filtered_for_current_user, lambda { |current_user|
    return all if current_user.admin_role?
    return visible if current_user.app_role?

    if current_user.editor_role?
      data_provider_ids = [current_user.data_provider_id] + User.restricted_role.map(&:data_provider_id)
      return includes(:poll).where(poll: { data_provider_id: data_provider_ids.compact.flatten })
    end

    includes(:poll).where(survey_polls: { data_provider_id: current_user.data_provider_id })
  }
end
```

Reply:
oder kann man nicht sogar:

```
def survey_comments
  object.comments.includes(:poll).filtered_for_current_user(context[:current_user])
end
```

das wär noch besser, denk ich.

Reply:
nee, reicht leider nicht, weil dann das spezielle `where(poll: { data_prov...` fehlt

Reply:
done in 97430de3f812161b8a8c47607b88ce80105e29c0

bitte überprüf das @marcometz, bevor wir mergen. scheint aber genau so zu funktionieren